components/PlanList/PlanList.js

```diff
@@ -1,28 +1,97 @@
-          <li>Dosage : {plan.dosage}</li>
-          <li>
-            Frequency : {plan.frequencyType} , {plan.frequencyTimes} , (
-            {plan.frequencyMornings ? "Mornings  " : " , "}
-            {plan.frequencyAfternoon ? "Afternoon  " : " , "}
-            {plan.frequencyEvening ? "Evenings  " : " , "}
-            {plan.frequencyBedTime ? "Bed Time  " : " "})
-          </li>
-          <li>Form : {plan.medicationForm}</li>
-          <li>Refill Date : {plan.refill}</li>
-          <li>Reason for taking : {plan.reason}</li>
-          <li>Instructions : {plan.instractions}</li>
-        </List>
-      ))}
+      <Title>My Medications Plan</Title>
+      <Span onClick={handleShowMyPlans}>
+        {showMyPlans ? "Hide Plans" : "Show Plans"} ({myPlans.length}) :
+      </Span>
+      {myPlans.length > 0 && showMyPlans && (
+        <>
+          {myPlans.map((plan, index) => (
+            <List key={index} importance={plan.importance}>
+              <li>Medication Name : {plan.medication_name}</li>
+              <li>Active Ingredients : {plan.active_ingredients}</li>
+              <li>Dosage : {plan.dosage}</li>
+              <li>
+                Frequency : {plan.frequencyType} , {plan.frequencyTimes} , (
+                {plan.frequencyMornings ? "Mornings  " : "   "}
+                {plan.frequencyAfternoon ? "Afternoon  " : "  "}
+                {plan.frequencyEvening ? "Evenings  " : "  "}
+                {plan.frequencyBedTime ? "Bed Time  " : "  "})
+              </li>
+              <li>Form : {plan.medicationForm}</li>
+              <li>Refill Date : {plan.refill}</li>
+              <li>Reason for taking : {plan.reason}</li>
+              <li>Instructions : {plan.instractions}</li>
+            </List>
+          ))}
+        </>
+      )}
+      {otherPlans.length > 0 && (
+        <>
+          <Title>Others Plans</Title>
+          {Object.values(
+            otherPlans.reduce((groupedPlans, plan) => {
+              const key = `${plan.name}-${plan.gender}-${plan.age}`;
+              if (!groupedPlans[key]) {
+                groupedPlans[key] = {
+                  name: plan.name,
+                  gender: plan.gender,
+                  age: plan.age,
+                  medications: [],
+                };
+              }
+              groupedPlans[key].medications.push(plan);
+              return groupedPlans;
+            }, {})
+          ).map((group, index) => (
+            <div key={index}>
+              <Heading>
+                {group.name} ({group.gender}, {group.age})
+              </Heading>
+              <Span onClick={() => handleShowPlan(index)}>
+                {showPlan === index ? "Hide Plans" : "Show Plans"} (
+                {group.medications.length}) :
+              </Span>
+              {showPlan === index &&
+                group.medications.map((plan, index) => (
+                  <List key={index} importance={plan.importance}>
```

Comment:
Here you should also not rely on the `index`.
Please add an `id` property to each medication on creation (when the form that creates medications is submitted) (e.g. using `uid()`).
Then use this id here:

```suggestion
                group.medications.map((plan) => (
                  <List key={plan.id} importance={plan.importance}>
```

Reply:
Thank you for your feedback , i have already changed it